tools/test/detect_targets_test.py

```diff
@@ -0,0 +1,85 @@
+"""
+mbed SDK
+Copyright (c) 2016 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+from collections import namedtuple
+from mock import patch, MagicMock
+from tools.detect_targets import get_interface_version
+from tools.test_api import get_autodetected_MUTS_list
+
+"""
+Tests for detect_targets.py
+"""
+
+class DetectTargetsTests(unittest.TestCase):
+    """
+    Test cases for Detect Target functionality
+    """
+
+    def setUp(self):
+        """
+        Called before each test case
+
+        :return:
+        """
+        # Gather a valid mount point from the host machine
+        muts = get_autodetected_MUTS_list()
+        
+        count = 0
+        
+        for mut in muts.values():
+            
+            count += 1
+            self.valid_mount_point = mut['disk']
+            break
+        
+        # If no targets are found, there is no way to determine Host OS mount point.
+        # Function should handle failure gracefully regardless of a mount point being present.
+        # Therefore it is safe to assume a valid mount point.
+        if count is 0:
+            self.valid_mount_point = "D:"        
```

Comment:
This is needed by the `test_interface_version_valid_mount_point` function, so it should be assigned unconditionally.

Reply:
How could we handle the case where there was no mount point (no device connected to the host), and across cases where host OS differs.. IE a valid mount point on Win 10 could be D: but a valid mount point in Linux is something like /media/{username}/DAPLink? There would need to be some condition that has to make an assumption at a certain point.

Reply:
> How could we handle the case where there was no mount point (no device connected to the host), and across cases where host OS differs.

Test 'em all?

> There would need to be some condition that has to make an assumption at a certain point.

Yes, but that assumption can be guaranteed with mocking.